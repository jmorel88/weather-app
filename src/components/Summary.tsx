interface Props {
  name: string;
  currentTemp: number;
  minTemp: number;
  maxTemp: number;
  description: string;
}

export default function Summary({
  name,
  currentTemp,
  minTemp,
  maxTemp,
  description,
}: Props) {
  return (
    <section data-testid="summary" className="p-8 bg-black-200 rounded-lg text-white-900 text-center md:text-left">
      <h2 className="text-4xl font-semibold mb-6">{name}</h2>
      <div className="md:flex items-end justify-between">
        <p>
          <span className="text-7xl text-orange font-bold inline-block md:inline mb-4 md:mb-0">
            {Math.round(currentTemp)}&deg;
          </span>
          <span className="text-xl md:text-2xl block md:inline mb-4">
            {description}
          </span>
        </p>
        <p className="text-xl md:text-2xl">
          Low {Math.round(minTemp)}&deg;
          <span className="inline-block mx-2" /> High {Math.round(maxTemp)}&deg;
        </p>
      </div>
    </section>
  );
}
