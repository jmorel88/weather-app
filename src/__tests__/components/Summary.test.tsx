import { describe, test } from "vitest";
import { render } from "@testing-library/react";
import Summary from "../../components/Summary";
import { mockWeatherResponse as weather } from "../mocks";

describe("Summary", () => {
  test("renders correctly", ({ expect }) => {
    const { container } = render(
      <Summary
        name={weather.name}
        minTemp={weather.main.temp_min}
        maxTemp={weather.main.temp_max}
        currentTemp={weather.main.temp}
        description={weather.weather[0].description}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
