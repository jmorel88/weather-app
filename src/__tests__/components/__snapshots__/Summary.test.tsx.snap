// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Summary > renders correctly 1`] = `
<div>
  <section
    class="p-8 bg-black-200 rounded-lg text-white-900 text-center md:text-left"
    data-testid="summary"
  >
    <h2
      class="text-4xl font-semibold mb-6"
    >
      New York
    </h2>
    <div
      class="md:flex items-end justify-between"
    >
      <p>
        <span
          class="text-7xl text-orange font-bold inline-block md:inline mb-4 md:mb-0"
        >
          55
          °
        </span>
        <span
          class="text-xl md:text-2xl block md:inline mb-4"
        >
          Cloudy
        </span>
      </p>
      <p
        class="text-xl md:text-2xl"
      >
        Low 
        50
        °
        <span
          class="inline-block mx-2"
        />
         High 
        60
        °
      </p>
    </div>
  </section>
</div>
`;
